action/protocol/execution/evm/evm.go

```diff
@@ -267,6 +298,24 @@ func ExecuteContract(
 	return retval, receipt, nil
 }
 
+func processSGD(ctx context.Context, sm protocol.StateManager, execution *action.Execution, consumedGas uint64, sgd SGDRegistry,
+) (address.Address, uint64, error) {
+	if execution.Contract() == action.EmptyAddress {
+		return nil, 0, nil
+	}
+
+	receiver, percentage, ok, err := sgd.CheckContract(ctx, execution.Contract())
```

Comment:
Pls provide the benchmark result for the change. Reading the contract directly for each tx in the block might cause performance regression. IMO, an indexer built for events emitted from the contract would be a better option